chia/server/server.py

```diff
@@ -267,13 +268,29 @@ async def start_server(self, on_connect: Callable = None):
         # this port from the socket itself and update self._port.
         self.site = web.TCPSite(
             self.runner,
-            host="0.0.0.0",
+            host="",  # should listen to both IPv4 and IPv6 on a dual-stack system
             port=int(self._port),
             shutdown_timeout=3,
             ssl_context=ssl_context,
         )
         await self.site.start()
-        self._port = self.runner.addresses[0][1]
+        #
+        # On a dual-stack system, we want to get the (first) IPv4 port unless
+        # prefer_ipv6 is set in which case we use the IPv6 port
+        #
+        if self._port == 0:
+            global_config = load_config(self.root_path, "config.yaml")
+            prefer_ipv6 = global_config.get("prefer_ipv6", False)
+            self._port = self.runner.addresses[0][1]  # sets the default to handle some edge cases
```

Comment:
Explain the corner cases?  Maybe do or maybe don't use it, but be aware of the search-and-find-or-default form of a for loop.  This way you don't pre-assign (maybe) wrong values.

```python
for x in y:
    if match(x):
        value = x
        break
else:
    # when we leave the loop without a break
    value = whatever
```

Reply:
Used this for/else syntax and added a utility function